subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -889,6 +891,252 @@ static int nrf_cloud_encode_service_info_ui(const struct nrf_cloud_svc_info_ui *
 	return 0;
 }
 
+#ifdef CONFIG_MODEM_INFO
+
+#define MODEM_INFO_JSON_KEY_NET_INF	"networkInfo"
+#define MODEM_INFO_JSON_KEY_SIM_INF	"simInfo"
+#define MODEM_INFO_JSON_KEY_DEV_INF	"deviceInfo"
+
+static int modem_info_add_data(struct lte_param *param, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	enum at_param_type data_type;
+	int ret;
+
+	__ASSERT_NO_MSG(param != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	memset(data_name, 0, ARRAY_SIZE(data_name));
+	ret = modem_info_name_get(param->type,
+				data_name);
+	if (ret < 0) {
+		LOG_DBG("Data name not obtained: %d", ret);
+		return -EINVAL;
+	}
+
+	data_type = modem_info_type_get(param->type);
+	if (data_type < 0) {
+		return -EINVAL;
+	}
+
+	if (data_type == AT_PARAM_TYPE_STRING &&
+	    param->type != MODEM_INFO_AREA_CODE) {
+		ret = json_add_str_cs(json_obj, data_name, param->value_string);
+		if (ret) {
+			return ret;
+		}
+	} else {
+		ret = json_add_num_cs(json_obj, data_name, param->value);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+static int modem_info_network_encode(struct network_param *network, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	int ret;
+
+	static const char lte_string[]	 = "LTE-M";
+	static const char nbiot_string[] = "NB-IoT";
+	static const char gps_string[]	 = " GPS";
+
+	__ASSERT_NO_MSG(network != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = modem_info_add_data(&network->current_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->sup_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->area_code, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->current_operator, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->ip_address, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->ue_mode, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_name_get(network->cellid_hex.type, data_name);
+	if (ret < 0) {
+		return ret;
+	}
+	data_name[ret] = '\0';
+
+	ret = json_add_num_cs(json_obj, data_name, network->cellid_dec);
+	if (ret) {
+		return ret;
+	}
+
+	network->network_mode[0] = '\0';
```

Comment:
could you remove this `network_mode` string from the `network_param` struct and just make it local to this function, that way we don't have to worry about the string size changing.

Reply:
✔